packages/lib/src/checkbox/Checkbox.tsx

```diff
@@ -1,115 +1,9 @@
-  (
-    {
-      checked,
-      defaultChecked = false,
-      value,
-      label = "",
-      labelPosition = "before",
-      name = "",
-      disabled = false,
-      optional = false,
-      readOnly = false,
-      onChange,
-      margin,
-      size = "fitContent",
-      tabIndex = 0,
-      ariaLabel = "Checkbox",
-    },
-    ref
-  ): JSX.Element => {
-    const labelId = `label-checkbox-${useId()}`;
-    const [innerChecked, setInnerChecked] = useState(defaultChecked);
-    const checkboxRef = useRef<HTMLSpanElement | null>(null);
-    const colorsTheme = useContext(HalstackContext);
-    const translatedLabels = useContext(HalstackLanguageContext);
-
-    const handleCheckboxChange = () => {
-      if (!disabled && !readOnly) {
-        if (document.activeElement !== checkboxRef.current) {
-          checkboxRef.current?.focus();
-        }
-        if (checked == null) {
-          setInnerChecked((innerCurrentlyChecked) => !innerCurrentlyChecked);
-        }
-        onChange?.(!(checked ?? innerChecked));
-      }
-    };
-
-    const handleKeyboard = (event: KeyboardEvent<HTMLSpanElement>) => {
-      switch (event.key) {
-        case " ":
-          event.preventDefault();
-          handleCheckboxChange();
-          break;
-        default:
-          break;
-      }
-    };
-
-    return (
-      <ThemeProvider theme={colorsTheme.checkbox}>
-        <MainContainer
-          disabled={disabled}
-          readOnly={readOnly}
-          onClick={handleCheckboxChange}
-          margin={margin}
-          size={size}
-          checked={checked ?? innerChecked}
-          ref={ref}
-        >
-          {label && (
-            <LabelContainer id={labelId} disabled={disabled} labelPosition={labelPosition} aria-label={label}>
-              {label}
-              {optional && ` ${translatedLabels.formFields.optionalLabel}`}
-            </LabelContainer>
-          )}
-          <ValueInput
-            type="checkbox"
-            checked={checked ?? innerChecked}
-            name={name}
-            value={value}
-            disabled={disabled}
-            readOnly
-          />
-          <CheckboxContainer>
-            <Checkbox
-              onKeyDown={handleKeyboard}
-              role="checkbox"
-              tabIndex={disabled ? -1 : tabIndex}
-              aria-checked={checked ?? innerChecked}
-              aria-disabled={disabled}
-              aria-readonly={readOnly}
-              aria-required={!disabled && !optional}
-              aria-labelledby={label ? labelId : undefined}
-              aria-label={label ? undefined : ariaLabel}
-              checked={checked ?? innerChecked}
-              disabled={disabled}
-              readOnly={readOnly}
-              ref={checkboxRef}
-            >
-              {(checked ?? innerChecked) && checkedIcon}
-            </Checkbox>
-          </CheckboxContainer>
-        </MainContainer>
-      </ThemeProvider>
-    );
-  }
-);
-
 const sizes = {
```

Comment:
`sizes`, `spaces`, `calculateWidth`, `getDisabledColor`, and `getEnabledColor` can be moved to a `utils.ts` file for cleaner code!